### DISCUSSION TITLE
FR:LVM support

### DISCUSSION THREAD

**Discussion by zzz929625** [nicolargo/glances#2800] [General]
I installed glances in proxmox to monitor the status of the system, everything is running fine except for not being able to monitor the lvm-thin pool, I looked through the documentation and googled the problem, but I didn't find a solution, I hope to be able to add the ability to monitor the lvm, or I didn't find it, and if I didn't find it I apologize for the lack of it.

**Comment by nicolargo** [comment]
Hi @anschein 

under the wood, Glances uses PSUtil to grab FileSystem information.

Can you please copy paste the result of the folowing command line:

```
./venv/bin/python
>>> import psutil
>>> psutil.disk_partitions(all=True)
...
```

**Reply by zzz929625** [reply]
[sdiskpart(device='sysfs', mountpoint='/sys', fstype='sysfs', opts='rw,nosuid,nodev,noexec,relatime', maxfile=255, maxpath=4096), sdiskpart(device='proc', mountpoint='/proc', fstype='proc', opts='rw,relatime', maxfile=255, maxpath=4096), sdiskpart(device='udev', mountpoint='/dev', fstype='devtmpfs', opts='rw,nosuid,relatime,size=65714660k,nr_inodes=16428665,mode=755,inode64', maxfile=255, maxpath=4096), sdiskpart(device='devpts', mountpoint='/dev/pts', fstype='devpts', opts='rw,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=000', maxfile=255, maxpath=4096), sdiskpart(device='tmpfs', mountpoint='/run', fstype='tmpfs', opts='rw,nosuid,nodev,noexec,relatime,size=13149800k,mode=755,inode64', maxfile=255, maxpath=4096), sdiskpart(device='rpool/ROOT/pve-1', mountpoint='/', fstype='zfs', opts='rw,relatime,xattr,noacl', maxfile=255, maxpath=4096), sdiskpart(device='securityfs', mountpoint='/sys/kernel/security', fstype='securityfs', opts='rw,nosuid,nodev,noexec,relatime', maxfile=255, maxpath=4096), sdiskpart(device='tmpfs', mountpoint='/dev/shm', fstype='tmpfs', opts='rw,nosuid,nodev,inode64', maxfile=255, maxpath=4096), sdiskpart(device='tmpfs', mountpoint='/run/lock', fstype='tmpfs', opts='rw,nosuid,nodev,noexec,relatime,size=5120k,inode64', maxfile=255, maxpath=4096), sdiskpart(device='cgroup2', mountpoint='/sys/fs/cgroup', fstype='cgroup2', opts='rw,nosuid,nodev,noexec,relatime', maxfile=255, maxpath=4096), sdiskpart(device='pstore', mountpoint='/sys/fs/pstore', fstype='pstore', opts='rw,nosuid,nodev,noexec,relatime', maxfile=255, maxpath=4096), sdiskpart(device='efivarfs', mountpoint='/sys/firmware/efi/efivars', fstype='efivarfs', opts='rw,nosuid,nodev,noexec,relatime', maxfile=255, maxpath=4096), sdiskpart(device='bpf', mountpoint='/sys/fs/bpf', fstype='bpf', opts='rw,nosuid,nodev,noexec,relatime,mode=700', maxfile=255, maxpath=4096), sdiskpart(device='systemd-1', mountpoint='/proc/sys/fs/binfmt_misc', fstype='autofs', opts='rw,relatime,fd=30,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=36359', maxfile=255, maxpath=4096), sdiskpart(device='mqueue', mountpoint='/dev/mqueue', fstype='mqueue', opts='rw,nosuid,nodev,noexec,relatime', maxfile=255, maxpath=4096), sdiskpart(device='hugetlbfs', mountpoint='/dev/hugepages', fstype='hugetlbfs', opts='rw,relatime,pagesize=2M', maxfile=255, maxpath=4096), sdiskpart(device='debugfs', mountpoint='/sys/kernel/debug', fstype='debugfs', opts='rw,nosuid,nodev,noexec,relatime', maxfile=255, maxpath=4096), sdiskpart(device='tracefs', mountpoint='/sys/kernel/tracing', fstype='tracefs', opts='rw,nosuid,nodev,noexec,relatime', maxfile=255, maxpath=4096), sdiskpart(device='fusectl', mountpoint='/sys/fs/fuse/connections', fstype='fusectl', opts='rw,nosuid,nodev,noexec,relatime', maxfile=255, maxpath=4096), sdiskpart(device='configfs', mountpoint='/sys/kernel/config', fstype='configfs', opts='rw,nosuid,nodev,noexec,relatime', maxfile=255, maxpath=4096), sdiskpart(device='ramfs', mountpoint='/run/credentials/systemd-sysusers.service', fstype='ramfs', opts='ro,nosuid,nodev,noexec,relatime,mode=700', maxfile=255, maxpath=4096), sdiskpart(device='ramfs', mountpoint='/run/credentials/systemd-tmpfiles-setup-dev.service', fstype='ramfs', opts='ro,nosuid,nodev,noexec,relatime,mode=700', maxfile=255, maxpath=4096), sdiskpart(device='ramfs', mountpoint='/run/credentials/systemd-sysctl.service', fstype='ramfs', opts='ro,nosuid,nodev,noexec,relatime,mode=700', maxfile=255, maxpath=4096), sdiskpart(device='rpool', mountpoint='/rpool', fstype='zfs', opts='rw,relatime,xattr,noacl', maxfile=255, maxpath=4096), sdiskpart(device='rpool/ROOT', mountpoint='/rpool/ROOT', fstype='zfs', opts='rw,relatime,xattr,noacl', maxfile=255, maxpath=4096), sdiskpart(device='rpool/data', mountpoint='/rpool/data', fstype='zfs', opts='rw,relatime,xattr,noacl', maxfile=255, maxpath=4096), sdiskpart(device='ramfs', mountpoint='/run/credentials/systemd-tmpfiles-setup.service', fstype='ramfs', opts='ro,nosuid,nodev,noexec,relatime,mode=700', maxfile=255, maxpath=4096), sdiskpart(device='binfmt_misc', mountpoint='/proc/sys/fs/binfmt_misc', fstype='binfmt_misc', opts='rw,nosuid,nodev,noexec,relatime', maxfile=255, maxpath=4096), sdiskpart(device='lxcfs', mountpoint='/var/lib/lxcfs', fstype='fuse.lxcfs', opts='rw,nosuid,nodev,relatime,user_id=0,group_id=0,allow_other', maxfile=255, maxpath=4096), sdiskpart(device='sunrpc', mountpoint='/run/rpc_pipefs', fstype='rpc_pipefs', opts='rw,relatime', maxfile=255, maxpath=4096), sdiskpart(device='/dev/fuse', mountpoint='/etc/pve', fstype='fuse', opts='rw,nosuid,nodev,relatime,user_id=0,group_id=0,default_permissions,allow_other', maxfile=256, maxpath=4096), sdiskpart(device='192.168.5.6:/backup', mountpoint='/mnt/pve/backup', fstype='nfs4', opts='rw,relatime,vers=4.1,rsize=1048576,wsize=1048576,namlen=255,hard,proto=tcp,timeo=600,retrans=2,sec=sys,clientaddr=192.168.5.7,local_lock=none,addr=192.168.5.6', maxfile=255, maxpath=4096), sdiskpart(device='tmpfs', mountpoint='/run/user/0', fstype='tmpfs', opts='rw,nosuid,nodev,relatime,size=13149796k,nr_inodes=3287449,mode=700,inode64', maxfile=255, maxpath=4096)]

**Reply by zzz929625** [reply]
<img width="369" alt="截屏2024-05-27 10 30 19" src="https://github.com/nicolargo/glances/assets/49552476/edc2fb16-2367-4544-9cc4-1a1a053421c2">
And this is my glances outputs for fs,I'm using glances4.0.6

**Comment by nicolargo** [comment]
By default, Glances only display physical device. If you want to enable logical mounts points, you need to add the file system type to the **allow** option in the [fs] section of the Glances configuration file:

Example:

```
[fs]
disable=False
hide=/boot.*,.*/snap.*
careful=50
warning=70
critical=90
allow=nfs4,zfs
```

**Reply by zzz929625** [reply]
I tried adding lvm to this config file, it didn't work, what characters do I need to add for lvm and lvm-thin?

**Reply by burgerga** [reply]
It's not supported by `psutil.disk_partitions`, so I guess also unsupported by glances

**Comment by nicolargo** [comment]
Try zsf
